src/main/java/appeng/container/implementations/ContainerNetworkStatus.java

```diff
@@ -167,39 +181,64 @@ public void detectAndSendChanges() {
 
             try {
                 final PacketMEInventoryUpdate piu = new PacketMEInventoryUpdate();
-
-                for (final Class<? extends IGridHost> machineClass : this.network.getMachinesClasses()) {
-                    final IItemList<IAEItemStack> list = AEApi.instance().storage().createItemList();
-                    for (final IGridNode machine : this.network.getMachines(machineClass)) {
-                        final IGridBlock blk = machine.getGridBlock();
-                        final ItemStack is = blk.getMachineRepresentation();
-                        if (is != null && is.getItem() != null) {
-                            final IAEItemStack ais = AEItemStack.create(is);
-                            ais.setStackSize(1);
-                            ais.setCountRequestable(
-                                    (long) PowerMultiplier.CONFIG.multiply(blk.getIdlePowerUsage() * 100.0));
-                            list.add(ais);
+                final IItemList<IAEItemStack> list = AEApi.instance().storage().createItemList();
+
+                // Network machine
+                if (this.isConsume) {
+                    for (final Class<? extends IGridHost> machineClass : this.network.getMachinesClasses()) {
+                        for (final IGridNode machine : this.network.getMachines(machineClass)) {
+                            final IGridBlock blk = machine.getGridBlock();
+                            final ItemStack is = blk.getMachineRepresentation();
+                            if (is != null && is.getItem() != null) {
+                                final IAEItemStack ais = AEItemStack.create(is);
+                                ais.setStackSize(1);
+                                ais.setCountRequestable(
+                                        (long) PowerMultiplier.CONFIG.multiply(blk.getIdlePowerUsage() * 100.0));
+                                list.add(ais);
+                            }
                         }
                     }
 
-                    for (final IAEItemStack ais : list) {
-                        piu.appendItem(ais);
+                } else {
+                    // Networ Cells
+                    final GridStorageCache sg = this.network.getCache(IStorageGrid.class);
+                    CellType selectedCellType = AEConfig.instance.selectedCellType();
+                    HashMap<ItemStack, Integer> cells = null;
+                    if (selectedCellType == CellType.ITEM) {
+                        cells = sg.getItemCells();
+                    } else if (selectedCellType == CellType.FLUID) {
+                        cells = sg.getFluidCells();
+                    } else if (selectedCellType == CellType.ESSENTIA) {
+                        cells = sg.getEssentiaCells();
+                    }
```

Comment:
Can be simplified using modern `switch` syntax:
```suggestion
                    ItemStackMap<Integer> cells = switch (selectedCellType) {
                        case ITEM -> sg.getItemCells();
                        case FLUID -> sg.getFluidCells();
                        case ESSENTIA -> sg.getEssentiaCells();
                    };
```